guppylang/checker/cfg_checker.py

```diff
@@ -117,18 +120,20 @@ def check_cfg(
     }
 
     # Finally, run the linearity check
-    check_cfg_linearity(checked_cfg)
+    from guppylang.checker.linearity_checker import check_cfg_linearity
```

Comment:
Why the local import here, instead of having it at the top?

Reply:
Circuilarity, since `linearity_checker.py` already depends on `cfg_checker.py`.

I could move `Signature`, `CheckedBB`, and `CheckedCFG` to `checker/core.py` to break the cycle if you prefer?